### ISSUE TITLE
ApplicationLoadBalancedFargateService: specify private subnets where the ALB will be deployed in

### ISSUE THREAD

**Issue by nuelee** [aws/aws-cdk#20775]
### Describe the feature

To have the option of manually selecting subnets for the ALB to be deployed in.
ApplicationLoadBalancedFargateService has this feature for tasks (taskSubnets) but not the ALB. 

Allowing ApplicationLoadBalancedFargateService to automatically deploy an ALB based on the vpc/cluster defined in the construct  works only when the VPC has few subnets (in my case 2 private subnets in different AZs) but doesn't work when you have multiple private subnets spread across multiple AZs hence returning the error below;

"A load balancer cannot be attached to multiple subnets in the same Availability Zone (Service: AmazonElasticLoadBalancing; Status Code: 400; Error Code: InvalidConfigurationRequest; Request ID: 67912baa-9955-455c-bdbb-f42870fc467e; Proxy: null)"
 

### Use Case

Fargate tasks are to be created in private subnet A (eu-west-2a) and private subnet B (eu-west-2b) hence require the
created ALB to be associated with  private subnet A (eu-west-2a) and private subnet B (eu-west-2b).

### Proposed Solution

Set ApplicationLoadBalancedFargateService so that it should either;
a. automatically assume the subnets selected in taskSubnets
    or
b. add an "ALB subnets" option that allows the a user to select the subnets similar to what we currenetly have for taskSubnets.

const loadBalancedFargateService = new ecsPatterns.ApplicationLoadBalancedFargateService(this, 
      'Service', {
            cluster,
            taskDefinition,
            desiredCount: 2,
            taskSubnets:{
              subnets:[
                ec2.Subnet.fromSubnetAttributes(this, 'subnet-12345678a', {
                  subnetId: 'subnet-12345678a', availabilityZone: 'eu-west-2a', routeTableId:'rtb-0001234'
                }),
                ec2.Subnet.fromSubnetAttributes(this, 'subnet-12345678b', {
                  subnetId: 'subnet-12345678a', availabilityZone: 'eu-west-2b', routeTableId:'rtb-000567'
                }),

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.28.0

### Environment details (OS name and version, etc.)

macOS version 12.4

**Comment by rupe120** [comment]
Yes please! I have a scenario where I am adding to an existing VPC and the subnet setup has multiple subnets in AZs and therefore errors and prevents me from being able to use `ApplicationLoadBalancedFargateService`

**Comment by twocoins-ca** [comment]
We need this new feature.

**Comment by Don-Tracksuit** [comment]
Is there a timeline on fixing this?

**Comment by lmanerich** [comment]
A possible workaround for those who are facing this issue is to remove the unwanted subnet from cdk.context.json
